src/SecTester.Core/Utils/StringUtils.cs

```diff
@@ -0,0 +1,66 @@
+using System.Text;
+
+namespace SecTester.Core.Utils;
+
+public static class StringUtils

```

Comment:
Why is it so complex?) It can be written in a few lines of code

```csharp
public static class StringUtils
{
  private static readonly Regex CamelCaseRegex = new(@"([a-z0-9])([A-Z])");
  private static readonly Regex PascalCaseRegex = new(@"([A-Z])([A-Z][a-z])");

  private static readonly Regex[] SupportedCaseRegexes =
  {
    PascalCaseRegex, CamelCaseRegex
  };

  public static string? ToSnakeCase(this string? value)
  {
    return string.IsNullOrEmpty(value)
      ? value
      : SupportedCaseRegexes.Aggregate(value!,
          (input, regex) => regex.Replace(input, "$1_$2"))
        .ToLowerInvariant();

  }
}
```